apps/dashboard/test/models/batch_connect/session_test.rb

```diff
@@ -504,4 +504,10 @@ def completed?
       assert_equal 1, base_dir.children.size
     end
   end
+
+  test 'can configure bc days old' do
+    with_modified_env({OOD_BC_DAYS_OLD: '3'}) do
+      assert 3, BatchConnect::Session.old_in_days
+    end
+  end
```

Comment:
I recalled as I typed it so I knew it'd be wrong. `OOD_BC_DAYS_OLD` which is wrong. Should be `OOD_BC_CARD_TIME`.  Tests didn't pick it up because it should have been `assert_equal`.

```suggestion
  test 'can configure bc days old' do
    with_modified_env({OOD_BC_CARD_TIME: '3'}) do
      assert_equal 3, BatchConnect::Session.old_in_days
    end
  end
```

But more than this 1 test case, I think we'd need coverage around these edge cases.

* default is 7.
* can configure it - which we have(ish)
* given a string (say `three` for example) - I'm OK with it returning 0 here actually.
* It should also be no less than 0. so a configuration of -1 results in 0 being returned.  I don't know what subtraction nightmares this may bring with negative numbers.

Reply:
Apologies for not noticing the test in detail myself for `OOD_BC_CARD_TIME ` variable. I should have been a bit careful with it.

For edge cases:
- Default will be 7.
- Can configure will also work.
- For the case in which we are returning 0, how will the UI behave? Will every session be considered old in that case? I was thinking we can just return 7(default) if the value comes out to be <= 0?

Reply:
> For the case in which we are returning 0, how will the UI behave?

The card should automatically be removed. That's how the behavior was before this feature to keep it existed.  Now if it can safely happen now? There could be a bug hiding here.

>  I was thinking we can just return 7(default) if the value comes out to be <= 0?

Now, I'm thinking negative's should return 0. The expectation of the user, who for whatever reason configured to delete things a week ago (-7) should get 0 to immediately delete.

I think the logic we're looking for is something like this:
```ruby
return 7 if nil or a string
return 0 if the_var.to_i <= 0

return the_var.to_i
```

Reply:
That makes sense!

I did test for configuration value 0. In that case, old? will always return true. I guess that's the expected behavior.

However, I believe there is one issue. Ruby's `to_i` method behaves differently in different scenarios:
```
'random string'.to_i => 0
'12random string'.to_i => 12
'0'.to_i => 1   # string 0
0.to_i => 0
```

So it is hard to differentiate between a 0 and a string. If we do it before `to_i` and check for string type before casting, then we miss out for integer strings like `'1'`, `'2'`. 

So I was thinking to use `Integer('0')` syntax for casting. It behaves like the following:
```
Integer('random string') => ArgumentError
Integer('0') => 0
Integer(0) => 0
Integer('-1') => -1
```

I believe this works well for our use case.

Reply:
Yea it looks like Ruby's best effort parsing. Honestly I'd rather have that behavior than a rescue.

Left a new comment. I'd like to avoid rescuing errors if possible. Provided something with a regular expression to catch the case where it's it's purely a string in a new comment.